testnet/stacks-node/src/nakamoto_node/signer_coordinator.rs

```diff
@@ -279,12 +279,21 @@ impl SignerCoordinator {
             }
         }
 
+        // build a BTreeMap of the various timeout steps
+        let mut block_rejection_timeout_steps = BTreeMap::<u64, Duration>::new();
+        for (percentage, duration) in self.block_rejection_timeout_steps.iter() {
+            let rejections_amount =
+                ((f64::from(self.total_weight) / 100.0) * f64::from(*percentage)) as u64;
+            block_rejection_timeout_steps.insert(rejections_amount, *duration);
+        }
```

Comment:
I still think this should be moved to `Self::new()` and then the struct should be updated to:

```rust
pub struct SignerCoordinator {
    ...
    /// The timeout configuration based on the percentage of rejections
    block_rejection_timeout_steps: BTreeMap<u64, Duration>,
}
```

Because `self.total_weight` doesn't ever change.